### PR TITLE
feat: Allows the user to provide the response after authentication fa…

### PR DESCRIPTION
**Pull request by zhcsyncer**
…iled, and interrupt executions of middleware chaining.

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

**Comment by czy88840616** [comment]
I think this place can use

```ts
throw new httpError.UnauthorizedError();
```

and use @Catch to custom reponse.

**Comment by zhcsyncer** [comment]
> I think this place can use
> 
> ```ts
> import { httpError } from '@midwayjs/core';
> 
> throw new httpError.UnauthorizedError();
> ```
> 
> and use @catch to custom reponse. like https://github.com/midwayjs/midway/blob/main/packages/upload/src/middleware.ts#L76

yes, it really is more suitable. 

But what is the `@catch`? You mean global exception handler?

**Comment by czy88840616** [comment]
Yes, it's means [this](https://midwayjs.org/docs/next/error_filter#%E5%BC%82%E5%B8%B8%E5%A4%84%E7%90%86%E5%99%A8) feature.

File: packages/passport/src/service/passport.ts [hunk 1]
```diff
@@ -148,6 +149,7 @@ export function PassportMiddleware(
                 return;
               } else {
                 res.status(401);
```
[the remark this answers is not in the document]
Comment (zhcsyncer) [reply packages/passport/src/service/passport.ts]: Ok,  I've seen [here](https://midwayjs.org/docs/next/error_filter#http-%E5%BC%82%E5%B8%B8) are handled by default and the `UnauthorizedError` already set status=**401**  in constructor.

**Comment by czy88840616** [comment]
LGTM

**Comment by zhcsyncer** [comment]
This change should be merge into the branch 2.x?

**Comment by czy88840616** [comment]
You should commit another pr for 2.x and use old code...

**Comment by zhcsyncer** [comment]
> You should commit another pr for 2.x and use old code...

Okay, I just noticed that the `main` is already in v3 and httpError is not exist... hhh

**Pull request merged**